tests/console/machinery.pm

```diff
@@ -13,13 +13,28 @@
 use base "consoletest";
 use strict;
 use testapi;
+use utils;
+use version_utils;
 
 sub run {
     select_console 'root-console';
-    assert_script_run 'which OneClickInstallCLI || zypper -n in yast2-metapackage-handler', 200;
-    my $ret_check = '[ "$?" = "141" ] || [ "$?" = "0" ]';
-    assert_script_run 'yes | OneClickInstallCLI http://machinery-project.org/machinery.ymp ; ' . $ret_check, 200;
+
+    if (is_sle) {
+        assert_script_run 'source /etc/os-release';
+        if (is_sle '>=15') {
+            assert_script_run 'SUSEConnect -p PackageHub/${VERSION_ID}/${CPU}';
+        }
+        else {
+            assert_script_run 'SUSEConnect -r ' . get_required_var('SCC_REGCODE');
+            assert_script_run 'SUSEConnect -p sle-module-adv-systems-management/12/${CPU}';
+        }
+    }
+
+    zypper_call 'in machinery';
```

Comment:
As discussed, if this fails I suggest to check for the corresponding exit code for unresolved dependencies and grep the log for the corresponding failure, e.g. (pseudo-code)

```
zypper_call('in machinery', exitcode => [0, $unresolved_dependency], log => "zypper.log");
return record_soft_failure 'bsc#…' if is_sle('>=15-sp1') && script_output <parse /var/log/zypper.log>, qr/rubygem(ruby:2.5.0:builder)/;
return record_soft_failure 'bsc#…' if is_sle('>=12-sp4') && …
}
```

What I checked locally:

```
isc co SUSE:SLE-15-SP1:GA 000product
cd $dir
grep 'ruby.*builder' *
```

result:

```
…
sle-ha-ftp-POOL-x86_64.kiwi:      <repopackage name="ruby2.5-rubygem-builder"/>
sle-ha-ftp-POOL-x86_64.kwd:ruby2.5-rubygem-builder: +Kwd:\nsupport_l3\n-Kwd:
…
```